### PR TITLE
feat: added aws-terragrunt-pr-plan workflow

### PR DESCRIPTION
**Pull request by jburns24**
Add some reusable workflows for running terragrunt on aws via OIDC auth

### CODE REVIEW

File: .github/workflows/aws-terragrunt-apply.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        uses: actions/checkout@v2
+
+      - name: Configure AWS Credentials via OIDC
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ${{ inputs.AWS_REGION }}
+          role-to-assume: ${{ inputs.AWS_OIDC_ROLE_ARN }}
+          role-session-name: github-action_aws-terragrunt-apply-oidc
+
+      - name: Apply infosec users
```
Comment (jknight-liatrio) [inline .github/workflows/aws-terragrunt-apply.yaml]: Nitpick but is this supposed to be used specifically for the infosec or is it supposed to be more general?

File: .github/workflows/aws-terragrunt-apply.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure AWS Credentials via OIDC
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ${{ inputs.AWS_REGION }}
+          role-to-assume: ${{ inputs.AWS_OIDC_ROLE_ARN }}
+          role-session-name: github-action_aws-terragrunt-apply-oidc
```
Comment (jknight-liatrio) [inline .github/workflows/aws-terragrunt-apply.yaml]: What's the scope of the session name here? How do these get stored between pipeline executions?

Comment (jknight-liatrio) [reply .github/workflows/aws-terragrunt-apply.yaml]: Is it more reusable if the `role-session-name` are `inputs`?

Comment (jknight-liatrio) [reply .github/workflows/aws-terragrunt-apply.yaml]: Does it make sense to add the `role-duration-seconds` here to limit the lifetime of the credential? 🤷

**Review by jknight-liatrio — APPROVED** [review]

File: .github/workflows/aws-terragrunt-apply.yaml (hunk 2 above)
Comment (jburns24) [reply .github/workflows/aws-terragrunt-apply.yaml]: I don't understand what you are asking in your first comment. 

Totally should have the role-session-name as an input good catch!

Doesn't hurt to add role-duration-seconds as another input here.

Comment (jburns24) [reply .github/workflows/aws-terragrunt-apply.yaml]: I think I understand the first comment now. I wanted to give the temporary creds an identifiable name so it is clear the credentials were requested by this reusable workflow. I am not sure how/if these get stored between pipeline executions. The thought was more if the request for these credentials gets logged anywhere on the aws side that the session name might help identify who/what was requesting them. 

That being said I do not know of any automatic logging that occurs on aws side. But I used this session name to help me validate things were working while developing

**Pull request merged**